com.ca.lsp.cobol/lsp-core-domain/src/test/java/com/broadcom/lsp/domain/cobol/databus/impl/DatabusEventSubscriptionE2ETest.java

```diff
@@ -0,0 +1,73 @@
+package com.broadcom.lsp.domain.cobol.databus.impl;
+
+import com.broadcom.lsp.domain.cobol.event.model.DataEvent;
+import com.broadcom.lsp.domain.cobol.event.model.DataEventType;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.fail;
+
+/**
+ * This class contains all the unit tests related to the pubblish/subscribe pattern handled by the
+ * databus.
+ */
+@Slf4j
+public class DatabusEventSubscriptionE2ETest extends DatabusConfigProvider {
```

Comment:
how do you value this test? if I delete one call from the tests they are still passing, what is the output here?

Reply:
This test verify that if the client subscribed to an event is also able to receive that event. I made one unit test for positive and negative scenarios, each of them test all the available events.